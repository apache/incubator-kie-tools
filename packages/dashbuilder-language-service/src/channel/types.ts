/*
 * Copyright 2022 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { TextDocument } from "vscode-languageserver-textdocument";
import { CompletionItemKind, Range } from "vscode-languageserver-types";

export declare type DashbuilderLsNodeType = "object" | "array" | "property" | "string" | "number" | "boolean" | "null";
export declare type DashbuilderJsonPath = (string | number)[];

/**
 * The AST node used in the LanguageServices
 */
export type DashbuilderLsNode = {
  type: DashbuilderLsNodeType;
  value?: any;
  offset: number;
  length: number;
  colonOffset?: number;
  parent?: DashbuilderLsNode;
  children?: DashbuilderLsNode[];
};

export interface TranslateArgs {
  completion: object | string;
  completionItemKind: CompletionItemKind;
  currentNodeRange?: Range;
  cursorOffset: number;
  document: TextDocument;
  overwriteRange?: Range;
}
