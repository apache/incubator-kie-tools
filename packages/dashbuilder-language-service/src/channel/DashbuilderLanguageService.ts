/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import {
  EditorLanguageService,
  EditorLanguageServiceArgs,
  ELsCompletionsMap,
  ELsNode,
  IEditorLanguageService,
} from "@kie-tools/json-yaml-language-service/dist/channel";
import { JSONSchema } from "vscode-json-languageservice";
import { TextDocument } from "vscode-languageserver-textdocument";
import { CodeLens, CompletionItem, Diagnostic, Position, Range } from "vscode-languageserver-types";
import {
  DashbuilderLanguageServiceCodeCompletion,
  DashbuilderLanguageServiceCodeCompletionFunctionsArgs,
} from "./DashbuilderLanguageServiceCodeCompletion";
import { DashbuilderLanguageServiceCodeLenses } from "./DashbuilderLanguageServiceCodeLenses";
import { CodeCompletionStrategy } from "./types";

export type DashbuilderLanguageServiceArgs = EditorLanguageServiceArgs;

export class DashbuilderLanguageService implements IEditorLanguageService {
  private readonly els: EditorLanguageService;

  constructor(private readonly args: DashbuilderLanguageServiceArgs) {
    this.els = new EditorLanguageService(this.args);
  }

  public async getCompletionItems(args: {
    content: string;
    uri: string;
    cursorPosition: Position;
    cursorWordRange: Range;
    rootNode: ELsNode | undefined;
    codeCompletionStrategy: CodeCompletionStrategy;
  }): Promise<CompletionItem[]> {
    return this.els.getCompletionItems({
      ...args,
      completions,
    });
  }

  public async getCodeLenses(args: {
    content: string;
    uri: string;
    rootNode: ELsNode | undefined;
    codeCompletionStrategy: CodeCompletionStrategy;
  }): Promise<CodeLens[]> {
    return this.els.getCodeLenses({
      ...args,
      codeLenses: DashbuilderLanguageServiceCodeLenses,
    });
  }

  public async getDiagnostics(args: {
    content: string;
    uriPath: string;
    rootNode: ELsNode | undefined;
    getSchemaDiagnostics: (args: { textDocument: TextDocument; fileMatch: string[] }) => Promise<Diagnostic[]>;
  }): Promise<Diagnostic[]> {
    return await this.els.getDiagnostics({
      ...args,
      getSchemaDiagnostics: args.getSchemaDiagnostics,
    });
  }

  public async getSchemaDiagnostics(args: {
    textDocument: TextDocument;
    fileMatch: string[];
    jsonSchema: JSONSchema;
  }): Promise<Diagnostic[]> {
    return this.els.getSchemaDiagnostics(args);
  }

  public dispose() {
    this.els.dispose();
  }
}

const completions: ELsCompletionsMap<DashbuilderLanguageServiceCodeCompletionFunctionsArgs> = new Map([
  [null, DashbuilderLanguageServiceCodeCompletion.getEmptyFileCodeCompletions],
]);
