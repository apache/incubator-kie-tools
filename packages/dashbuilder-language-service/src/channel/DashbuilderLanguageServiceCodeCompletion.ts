/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import {
  EditorLanguageServiceCodeCompletionFunctions,
  EditorLanguageServiceCodeCompletionFunctionsArgs,
  EditorLanguageServiceEmptyFileCodeCompletionFunctionArgs,
} from "@kie-tools/json-yaml-language-service/dist/channel";
import { CompletionItem, CompletionItemKind, InsertTextFormat, Range } from "vscode-languageserver-types";
import { dashbuilderCompletion } from "../assets/code-completions/";

export type DashbuilderLanguageServiceCodeCompletionFunctionsArgs =
  EditorLanguageServiceCodeCompletionFunctionsArgs & {};

/**
 * DashbuilderLanguageService CodeCompletion functions
 */
export const DashbuilderLanguageServiceCodeCompletion: EditorLanguageServiceCodeCompletionFunctions = {
  getEmptyFileCodeCompletions(
    args: EditorLanguageServiceEmptyFileCodeCompletionFunctionArgs
  ): Promise<CompletionItem[]> {
    const kind = CompletionItemKind.Text;
    const label = "Create your first dashboard";
    return Promise.resolve([
      {
        kind,
        label,
        detail: "Start with a simple dashboard",
        sortText: `100_${label}`, //place the completion on top in the menu
        textEdit: {
          newText: args.codeCompletionStrategy.translate({
            ...args,
            completion: dashbuilderCompletion,
            completionItemKind: kind,
          }),
          range: Range.create(args.cursorPosition, args.cursorPosition),
        },
        insertTextFormat: InsertTextFormat.Snippet,
      },
    ]);
  },
};
