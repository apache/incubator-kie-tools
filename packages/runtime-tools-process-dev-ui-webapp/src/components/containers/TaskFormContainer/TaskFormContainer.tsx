/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from "react";
import { componentOuiaProps, OUIAProps } from "@kogito-apps/ouia-tools/dist/utils/OuiaUtils";
import { GraphQL } from "@kogito-apps/consoles-common/dist/graphql";
import UserTaskInstance = GraphQL.UserTaskInstance;
import { EmbeddedTaskForm } from "@kogito-apps/task-form";
import { Form } from "@kogito-apps/components-common/dist/types";
import { useTaskFormGatewayApi } from "../../../channel/TaskForms/TaskFormContext";
import { useDevUIAppContext } from "../../contexts/DevUIAppContext";

interface Props {
  userTask: UserTaskInstance;
  onSubmitSuccess: (message: string) => void;
  onSubmitError: (message: string, details?: string) => void;
}

const TaskFormContainer: React.FC<Props & OUIAProps> = ({
  userTask,
  onSubmitSuccess,
  onSubmitError,
  ouiaId,
  ouiaSafe,
}) => {
  const gatewayApi = useTaskFormGatewayApi();
  const appContext = useDevUIAppContext();

  return (
    <EmbeddedTaskForm
      {...componentOuiaProps(ouiaId, "task-form-container", ouiaSafe)}
      userTask={userTask}
      user={appContext.getCurrentUser()}
      driver={{
        doSubmit(phase?: string, payload?: any): Promise<any> {
          return new Promise<any>((resolve, reject) => {
            gatewayApi
              .doSubmit(userTask, phase, payload)
              .then((response) => {
                onSubmitSuccess(phase);
                resolve(response);
              })
              .catch((error) => {
                const message = error.response ? error.response.data : error.message;
                onSubmitError(phase, message);
                reject(error);
              });
          });
        },
        getTaskFormSchema(): Promise<Record<string, any>> {
          return gatewayApi.getTaskFormSchema(userTask);
        },
        getCustomForm(): Promise<Form> {
          return gatewayApi.getCustomForm(userTask);
        },
      }}
      targetOrigin={appContext.getDevUIUrl()}
    />
  );
};

export default TaskFormContainer;
