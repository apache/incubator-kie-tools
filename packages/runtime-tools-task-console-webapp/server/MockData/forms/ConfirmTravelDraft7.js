/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
module.exports = ConfirmTravelFormDraft7 = {
  $schema: "http://json-schema.org/draft-07/schema#",
  type: "object",
  definitions: {
    Address: {
      type: "object",
      properties: {
        street: {
          type: "string",
        },
        city: {
          type: "string",
        },
        zipCode: {
          type: "string",
        },
        country: {
          type: "string",
        },
      },
    },
    Flight: {
      type: "object",
      properties: {
        flightNumber: {
          type: "string",
        },
        seat: {
          type: "string",
        },
        gate: {
          type: "string",
        },
        departure: {
          type: "string",
          format: "date-time",
        },
        arrival: {
          type: "string",
          format: "date-time",
        },
      },
    },
    Hotel: {
      type: "object",
      properties: {
        name: {
          type: "string",
        },
        address: {
          $ref: "#/definitions/Address",
        },
        phone: {
          type: "string",
        },
        bookingNumber: {
          type: "string",
        },
        room: {
          type: "string",
        },
      },
    },
  },
  properties: {
    flight: {
      allOf: [{ $ref: "#/definitions/Flight" }, { input: true }],
    },
    hotel: {
      allOf: [{ $ref: "#/definitions/Hotel" }, { input: true }, { output: true }],
    },
  },
  phases: ["complete", "release"],
};
