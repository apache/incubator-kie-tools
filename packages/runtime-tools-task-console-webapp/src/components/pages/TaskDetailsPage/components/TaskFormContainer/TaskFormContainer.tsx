/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from "react";
import { OUIAProps, componentOuiaProps } from "@kie-tools/runtime-tools-components/dist/ouiaTools";
import { useKogitoAppContext } from "@kie-tools/runtime-tools-components/dist/contexts/KogitoAppContext";
import { Form } from "@kie-tools/runtime-tools-shared-gateway-api/dist/types";
import { EmbeddedTaskForm } from "@kie-tools/runtime-tools-process-enveloped-components/dist/taskForm";
import { UserTaskInstance } from "@kie-tools/runtime-tools-process-gateway-api/dist/types";
import { useTaskFormGatewayApi } from "@kie-tools/runtime-tools-process-webapp-components/dist/TaskForms";

interface Props {
  userTask: UserTaskInstance;
  onSubmitSuccess: (message: string) => void;
  onSubmitError: (message: string, details?: string) => void;
}

const TaskFormContainer: React.FC<Props & OUIAProps> = ({
  userTask,
  onSubmitSuccess,
  onSubmitError,
  ouiaId,
  ouiaSafe,
}) => {
  const gatewayApi = useTaskFormGatewayApi();
  const kogitoAppContext = useKogitoAppContext();

  return (
    <EmbeddedTaskForm
      {...componentOuiaProps(ouiaId, "task-form-container", ouiaSafe)}
      userTask={userTask}
      user={kogitoAppContext.getCurrentUser()}
      driver={{
        doSubmit(phase?: string, payload?: any): Promise<any> {
          return gatewayApi
            .doSubmit(userTask, phase, payload)
            .then((result) => onSubmitSuccess(phase))
            .catch((error) => {
              const message = error.response ? error.response.data : error.message;
              onSubmitError(phase, message);
            });
        },
        getTaskFormSchema(): Promise<Record<string, any>> {
          return gatewayApi.getTaskFormSchema(userTask);
        },
        getCustomForm(): Promise<Form> {
          return gatewayApi.getCustomForm(userTask);
        },
      }}
      targetOrigin={window.location.origin}
    />
  );
};

export default TaskFormContainer;
