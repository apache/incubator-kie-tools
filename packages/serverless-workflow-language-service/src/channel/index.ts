export * from "./SwfJsonLanguageService";
export * from "./SwfLanguageService";
export * from "./SwfYamlLanguageService";
