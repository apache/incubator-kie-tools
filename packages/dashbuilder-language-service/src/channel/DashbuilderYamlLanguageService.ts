/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import {
  EditorYamlCodeCompletionStrategy,
  EditorYamlLanguageService,
  ELsNode,
  ELsShouldCreateCodelensArgs,
  IEditorLanguageService,
  indentText,
  positions_equals,
  ShouldCompleteArgs,
  TranslateArgs,
} from "@kie-tools/json-yaml-language-service/dist/channel";
import { TextDocument } from "vscode-languageserver-textdocument";
import { CodeLens, CompletionItem, CompletionItemKind, Diagnostic, Position, Range } from "vscode-languageserver-types";
import { dump } from "yaml-language-server-parser";
import { FileLanguage } from "../api";
import { DASHBUILDER_SCHEMA } from "../assets/schemas";
import { DashbuilderLanguageService } from "./DashbuilderLanguageService";
import { CodeCompletionStrategy, ShouldCreateCodelensArgs } from "./types";

export class DashbuilderYamlLanguageService implements IEditorLanguageService {
  private readonly ls: DashbuilderLanguageService;
  private readonly yamlELs: EditorYamlLanguageService;
  private readonly codeCompletionStrategy: DashbuilderYamlCodeCompletionStrategy;

  constructor() {
    this.ls = new DashbuilderLanguageService({
      fs: {},
      lang: {
        fileLanguage: FileLanguage.YAML,
        fileMatch: ["*.dash.yaml", "*.dash.yml"],
      },
    });
    this.codeCompletionStrategy = new DashbuilderYamlCodeCompletionStrategy();
    this.yamlELs = new EditorYamlLanguageService({
      ls: this.ls,
      codeCompletionStrategy: this.codeCompletionStrategy,
    });
  }

  public async getCompletionItems(args: {
    content: string;
    uri: string;
    cursorPosition: Position;
    cursorWordRange: Range;
  }): Promise<CompletionItem[]> {
    return await this.yamlELs.getCompletionItems(args);
  }

  public async getCodeLenses(args: { content: string; uri: string }): Promise<CodeLens[]> {
    return await this.yamlELs.getCodeLenses(args);
  }

  public async getDiagnostics(args: { content: string; uriPath: string }): Promise<Diagnostic[]> {
    return await this.yamlELs.getDiagnostics({
      ...args,
      getSchemaDiagnostics: (args: { textDocument: TextDocument; fileMatch: string[] }) =>
        this.getSchemaDiagnostics(args),
    });
  }

  public async getSchemaDiagnostics(args: { textDocument: TextDocument; fileMatch: string[] }): Promise<Diagnostic[]> {
    return await this.yamlELs.getSchemaDiagnostics({
      ...args,
      jsonSchema: DASHBUILDER_SCHEMA,
    });
  }

  public dispose() {
    return this.yamlELs.dispose();
  }
}

export class DashbuilderYamlCodeCompletionStrategy implements CodeCompletionStrategy {
  private eLsCodeCompletionStrategy: EditorYamlCodeCompletionStrategy;

  constructor() {
    this.eLsCodeCompletionStrategy = new EditorYamlCodeCompletionStrategy();
  }

  public translate(args: TranslateArgs): string {
    const completionDump = dump(args.completion, {}).slice(2, -1).trim();
    if (["{}", "[]"].includes(completionDump) || args.completionItemKind === CompletionItemKind.Text) {
      return completionDump;
    }
    const skipFirstLineIndent = args.completionItemKind !== CompletionItemKind.Module;
    const completionItemNewLine = args.completionItemKind === CompletionItemKind.Module ? "\n" : "";
    const completionText = completionItemNewLine + indentText(completionDump, 2, " ", skipFirstLineIndent);
    return ([CompletionItemKind.Interface, CompletionItemKind.Reference] as CompletionItemKind[]).includes(
      args.completionItemKind
    ) && positions_equals(args.overwriteRange?.start ?? null, args.currentNodeRange?.start ?? null)
      ? `- ${completionText}\n`
      : completionText;
  }

  public formatLabel(label: string, completionItemKind: CompletionItemKind): string {
    return this.eLsCodeCompletionStrategy.formatLabel(label, completionItemKind);
  }

  public getStartNodeValuePosition(document: TextDocument, node: ELsNode): Position | undefined {
    return this.eLsCodeCompletionStrategy.getStartNodeValuePosition(document, node);
  }

  public shouldComplete(args: ShouldCompleteArgs): boolean {
    return this.eLsCodeCompletionStrategy.shouldComplete(args);
  }

  public shouldCreateCodelens(args: ShouldCreateCodelensArgs): boolean {
    return this.eLsCodeCompletionStrategy.shouldCreateCodelens(args as ELsShouldCreateCodelensArgs);
  }
}
