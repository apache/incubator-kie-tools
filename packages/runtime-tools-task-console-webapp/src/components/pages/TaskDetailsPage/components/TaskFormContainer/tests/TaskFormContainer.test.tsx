/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from "react";
import { mount } from "enzyme";
import TaskFormContainer from "../TaskFormContainer";
import { TaskFormGatewayApi } from "../../../../../../channel/forms";
import * as TaskFormContext from "../../../../../../channel/forms/TaskFormContext";
import { UserTaskInstance } from "@kogito-apps/task-console-shared";
import { KogitoAppContextProvider } from "@kogito-apps/consoles-common/dist/environment/context";
import { DefaultUser, User } from "@kogito-apps/consoles-common/dist/environment/auth";

const testUserTask: UserTaskInstance = {
  id: "45a73767-5da3-49bf-9c40-d533c3e77ef3",
  description: null,
  name: "VisaApplication",
  referenceName: "Apply for visa",
  priority: "1",
  processInstanceId: "9ae7ce3b-d49c-4f35-b843-8ac3d22fa427",
  processId: "travels",
  rootProcessInstanceId: null,
  rootProcessId: null,
  state: "Ready",
  actualOwner: null,
  adminGroups: [],
  adminUsers: [],
  completed: null,
  started: new Date("2020-02-19T11:11:56.282Z"),
  excludedUsers: [],
  potentialGroups: [],
  potentialUsers: [],
  inputs:
    '{"Skippable":"true","trip":{"city":"Boston","country":"US","begin":"2020-02-19T23:00:00.000+01:00","end":"2020-02-26T23:00:00.000+01:00","visaRequired":true},"TaskName":"VisaApplication","NodeName":"Apply for visa","traveller":{"firstName":"Rachel","lastName":"White","email":"rwhite@gorle.com","nationality":"Polish","address":{"street":"Cabalone","city":"Zerf","zipCode":"765756","country":"Poland"}},"Priority":"1"}',
  outputs: "{}",
  lastUpdate: new Date("2020-02-19T11:11:56.282Z"),
  endpoint:
    "http://localhost:8080/travels/9ae7ce3b-d49c-4f35-b843-8ac3d22fa427/VisaApplication/45a73767-5da3-49bf-9c40-d533c3e77ef3",
};

const MockTaskFormGatewayApi = jest.fn<TaskFormGatewayApi, []>(() => ({
  getTaskFormSchema: jest.fn(),
  getCustomForm: jest.fn(),
  doSubmit: jest.fn(),
}));

jest.spyOn(TaskFormContext, "useTaskFormGatewayApi").mockImplementation(() => new MockTaskFormGatewayApi());

const user: User = new DefaultUser("jon", []);

describe("TaskFormContainer tests", () => {
  it("Snapshot", () => {
    const onSubmit = jest.fn();
    const onFailure = jest.fn();
    const wrapper = mount(
      <KogitoAppContextProvider
        userContext={{
          getCurrentUser: () => user,
        }}
      >
        <TaskFormContainer userTask={testUserTask} onSubmitSuccess={onSubmit} onSubmitError={onFailure} />
      </KogitoAppContextProvider>
    ).find("TaskFormContainer");

    expect(wrapper).toMatchSnapshot();

    const forwardRef = wrapper.childAt(0);
    expect(forwardRef.props().driver).not.toBeNull();
    expect(forwardRef.props().user).toStrictEqual(user);
    expect(forwardRef.props().targetOrigin).toBe("http://localhost");
  });
});
