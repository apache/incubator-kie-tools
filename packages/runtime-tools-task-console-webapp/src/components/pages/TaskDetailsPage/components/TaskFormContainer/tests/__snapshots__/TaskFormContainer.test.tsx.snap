// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskFormContainer tests Snapshot 1`] = `
<TaskFormContainer
  onSubmitError={[MockFunction]}
  onSubmitSuccess={[MockFunction]}
  userTask={
    Object {
      "actualOwner": null,
      "adminGroups": Array [],
      "adminUsers": Array [],
      "completed": null,
      "description": null,
      "endpoint": "http://localhost:8080/travels/9ae7ce3b-d49c-4f35-b843-8ac3d22fa427/VisaApplication/45a73767-5da3-49bf-9c40-d533c3e77ef3",
      "excludedUsers": Array [],
      "id": "45a73767-5da3-49bf-9c40-d533c3e77ef3",
      "inputs": "{\\"Skippable\\":\\"true\\",\\"trip\\":{\\"city\\":\\"Boston\\",\\"country\\":\\"US\\",\\"begin\\":\\"2020-02-19T23:00:00.000+01:00\\",\\"end\\":\\"2020-02-26T23:00:00.000+01:00\\",\\"visaRequired\\":true},\\"TaskName\\":\\"VisaApplication\\",\\"NodeName\\":\\"Apply for visa\\",\\"traveller\\":{\\"firstName\\":\\"Rachel\\",\\"lastName\\":\\"White\\",\\"email\\":\\"rwhite@gorle.com\\",\\"nationality\\":\\"Polish\\",\\"address\\":{\\"street\\":\\"Cabalone\\",\\"city\\":\\"Zerf\\",\\"zipCode\\":\\"765756\\",\\"country\\":\\"Poland\\"}},\\"Priority\\":\\"1\\"}",
      "lastUpdate": 2020-02-19T11:11:56.282Z,
      "name": "VisaApplication",
      "outputs": "{}",
      "potentialGroups": Array [],
      "potentialUsers": Array [],
      "priority": "1",
      "processId": "travels",
      "processInstanceId": "9ae7ce3b-d49c-4f35-b843-8ac3d22fa427",
      "referenceName": "Apply for visa",
      "rootProcessId": null,
      "rootProcessInstanceId": null,
      "started": 2020-02-19T11:11:56.282Z,
      "state": "Ready",
    }
  }
>
  <ForwardRef
    data-ouia-component-type="task-form-container"
    data-ouia-safe={true}
    driver={
      Object {
        "doSubmit": [Function],
        "getCustomForm": [Function],
        "getTaskFormSchema": [Function],
      }
    }
    targetOrigin="http://localhost"
    user={
      DefaultUser {
        "groups": Array [],
        "id": "jon",
      }
    }
    userTask={
      Object {
        "actualOwner": null,
        "adminGroups": Array [],
        "adminUsers": Array [],
        "completed": null,
        "description": null,
        "endpoint": "http://localhost:8080/travels/9ae7ce3b-d49c-4f35-b843-8ac3d22fa427/VisaApplication/45a73767-5da3-49bf-9c40-d533c3e77ef3",
        "excludedUsers": Array [],
        "id": "45a73767-5da3-49bf-9c40-d533c3e77ef3",
        "inputs": "{\\"Skippable\\":\\"true\\",\\"trip\\":{\\"city\\":\\"Boston\\",\\"country\\":\\"US\\",\\"begin\\":\\"2020-02-19T23:00:00.000+01:00\\",\\"end\\":\\"2020-02-26T23:00:00.000+01:00\\",\\"visaRequired\\":true},\\"TaskName\\":\\"VisaApplication\\",\\"NodeName\\":\\"Apply for visa\\",\\"traveller\\":{\\"firstName\\":\\"Rachel\\",\\"lastName\\":\\"White\\",\\"email\\":\\"rwhite@gorle.com\\",\\"nationality\\":\\"Polish\\",\\"address\\":{\\"street\\":\\"Cabalone\\",\\"city\\":\\"Zerf\\",\\"zipCode\\":\\"765756\\",\\"country\\":\\"Poland\\"}},\\"Priority\\":\\"1\\"}",
        "lastUpdate": 2020-02-19T11:11:56.282Z,
        "name": "VisaApplication",
        "outputs": "{}",
        "potentialGroups": Array [],
        "potentialUsers": Array [],
        "priority": "1",
        "processId": "travels",
        "processInstanceId": "9ae7ce3b-d49c-4f35-b843-8ac3d22fa427",
        "referenceName": "Apply for visa",
        "rootProcessId": null,
        "rootProcessInstanceId": null,
        "started": 2020-02-19T11:11:56.282Z,
        "state": "Ready",
      }
    }
  >
    <ForwardRef(RefForwardingEmbeddedEnvelope)
      apiImpl={
        EmbeddedTaskFormChannelApiImpl {
          "driver": Object {
            "doSubmit": [Function],
            "getCustomForm": [Function],
            "getTaskFormSchema": [Function],
          },
        }
      }
      config={
        Object {
          "containerType": "div",
        }
      }
      origin="http://localhost"
      pollInit={[Function]}
      refDelegate={[Function]}
    >
      <div />
    </ForwardRef(RefForwardingEmbeddedEnvelope)>
  </ForwardRef>
</TaskFormContainer>
`;
